runtime/connectors/s3/s3.go

```diff
@@ -72,46 +77,49 @@ func (c connector) Spec() connectors.Spec {
 	return spec
 }
 
-func (c connector) ConsumeAsFile(ctx context.Context, env *connectors.Env, source *connectors.Source) (string, error) {
+func (c connector) ConsumeAsFile(ctx context.Context, env *connectors.Env, source *connectors.Source) ([]string, error) {
 	conf, err := ParseConfig(source.Properties)
 	if err != nil {
-		return "", fmt.Errorf("failed to parse config: %w", err)
+		return nil, fmt.Errorf("failed to parse config: %w", err)
 	}
 
-	// The session the S3 Downloader will use
-	sess, err := getAwsSessionConfig(conf)
-	if err != nil {
-		return "", fmt.Errorf("failed to start session: %w", err)
+	if !doublestar.ValidatePattern(conf.Path) {
+		// ideally this should be validated at much earlier stage
+		// keeping it here to have s3 specific validations
+		return nil, fmt.Errorf("glob pattern %s is invalid", conf.Path)
 	}
```

Comment:
Do the validation in the s3 connector's `ParseConfig` instead. It might be used for earlier validation.